include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -296,6 +296,12 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _Up __u, _LRp __brick_lea
     // distribution is ~1 work groups per compute init
     if (__exec.queue().get_device().is_cpu())
         __iters_per_work_item = (__n - 1) / (__max_compute_units * __work_group_size) + 1;
+    // __iters_per_work_item shows number of elements to reduce on global memory
```

Comment:
Probably, it make sense to move "__iters_per_work_item" calculation (or whole "work group size" calculation stuff) into an utility function/class..
Is such calculation applicable for the other SYCL backend patterns?

Reply:
"__iters_per_work_item" has unique meaning for reduction pattern.

Though, it seems possible to generalize this code (it occur in find, reduce and scan):
```c++
    __work_group_size = ...__max_work_group_size(...);
    // change __work_group_size according to local memory limit
    __work_group_size = ...__max_local_allocation_size(...);
#if _ONEDPL_COMPILE_KERNEL
    sycl::kernel __kernel = ...__compile_kernel(...);
    __work_group_size = ::std::min(__work_group_size, ...__kernel_work_group_size(...));
#endif
```

Reply:
Added TODO comment